### PR TITLE
Added config parameter for irods connection pool size. 

### PR DESCRIPTION
**Pull request by kxk302**
Also, debugging the size of active/idle connections.

### CODE REVIEW

**Comment by kxk302** [comment]
There are 3 changes in this PR:

1) The irods connections were not being cleaned up properly (connection cleanup was in destructor and was not called). Per Mariius' suggestion, moved it to shutdown method. This was causing error messages in irods log on Test.

2) Added irods connection pool size as a parameter. 

3) Added timer statements for all irods methods.

File: lib/galaxy/objectstore/irods.py [hunk 1]
```diff
@@ -199,28 +206,42 @@ def __init__(self, config, config_dict):
         self.extra_dirs.update(extra_dirs)
 
         self._initialize()
+        log.debug("irods __init__ %s", reload_timer)
 
-    def __del__(self):
+    def shutdown(self):
+        # This call will cleanup all the connections in the connection pool
+        log.debug("In __del__ Number of active connections: %s, Number of idle connections: %s", len(self.session.pool.active), len(self.session.pool.idle))
```
Comment (mvdbeek) [inline lib/galaxy/objectstore/irods.py]:
```suggestion
        log.debug("In __shutdown__ Number of active connections: %s, Number of idle connections: %s", len(self.session.pool.active), len(self.session.pool.idle))
```

**Review by mvdbeek — DISMISSED** [review]
Thanks @kxk302!

**Comment by mvdbeek** [comment]
Looks like there is a valid error in the upload datatype test:
```
2020-05-20T20:15:37.5086569Z ==================================== ERRORS ====================================
2020-05-20T20:15:37.5088322Z __ ERROR at teardown of test_upload_datatype_dos_irods_and_disk[affy_v_3.cel] __
2020-05-20T20:15:37.5089413Z 
2020-05-20T20:15:37.5092244Z cls = <class 'integration.objectstore.test_objectstore_datatype_upload.UploadTestDosIrodsAndDiskTestCase'>
2020-05-20T20:15:37.5092597Z 
2020-05-20T20:15:37.5092988Z     @classmethod
2020-05-20T20:15:37.5094546Z     def tearDownClass(cls):
2020-05-20T20:15:37.5094796Z         stop_irods(cls.container_name)
2020-05-20T20:15:37.5095031Z >       super(UploadTestDatatypeDataTestCase, cls).tearDownClass()
2020-05-20T20:15:37.5095241Z 
2020-05-20T20:15:37.5095824Z test/integration/objectstore/test_objectstore_datatype_upload.py:142: 
2020-05-20T20:15:37.5096688Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-05-20T20:15:37.5097375Z lib/galaxy_test/driver/integration_util.py:95: in tearDownClass
2020-05-20T20:15:37.5097768Z     cls._test_driver.tear_down()
2020-05-20T20:15:37.5098397Z lib/galaxy_test/driver/driver_util.py:859: in tear_down
2020-05-20T20:15:37.5098761Z     self.stop_servers()
2020-05-20T20:15:37.5099395Z lib/galaxy_test/driver/driver_util.py:865: in stop_servers
2020-05-20T20:15:37.5099783Z     server_wrapper.stop()
2020-05-20T20:15:37.5100396Z lib/galaxy_test/driver/driver_util.py:699: in stop
2020-05-20T20:15:37.5100949Z     self._app.shutdown()
2020-05-20T20:15:37.5101722Z lib/galaxy/app.py:323: in shutdown
2020-05-20T20:15:37.5102085Z     raise exception
2020-05-20T20:15:37.5102657Z lib/galaxy/app.py:294: in shutdown
2020-05-20T20:15:37.5103071Z     self.object_store.shutdown()
2020-05-20T20:15:37.5103701Z lib/galaxy/objectstore/__init__.py:816: in shutdown
2020-05-20T20:15:37.5104523Z     super(DistributedObjectStore, self).shutdown()
2020-05-20T20:15:37.5105228Z lib/galaxy/objectstore/__init__.py:619: in shutdown
2020-05-20T20:15:37.5105918Z     store.shutdown()
2020-05-20T20:15:37.5106517Z lib/galaxy/objectstore/irods.py:214: in shutdown
2020-05-20T20:15:37.5106889Z     self.session.cleanup()
2020-05-20T20:15:37.5107928Z .venv/lib/python3.7/site-packages/irods/session.py:43: in cleanup
2020-05-20T20:15:37.5108130Z     conn.disconnect()
2020-05-20T20:15:37.5110157Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-05-20T20:15:37.5110273Z 
2020-05-20T20:15:37.5112295Z self = <irods.connection.Connection object at 0x7f53a6e49ed0>
2020-05-20T20:15:37.5112516Z 
2020-05-20T20:15:37.5112721Z     def disconnect(self):
2020-05-20T20:15:37.5113242Z         disconnect_msg = iRODSMessage(msg_type='RODS_DISCONNECT')
2020-05-20T20:15:37.5113494Z         self.send(disconnect_msg)
2020-05-20T20:15:37.5113741Z         try:
2020-05-20T20:15:37.5113955Z             # SSL shutdown handshake
2020-05-20T20:15:37.5114165Z             self.socket = self.socket.unwrap()
2020-05-20T20:15:37.5114709Z         except AttributeError:
2020-05-20T20:15:37.5114930Z             pass
2020-05-20T20:15:37.5115109Z >       self.socket.shutdown(socket.SHUT_RDWR)
2020-05-20T20:15:37.5115310Z E       OSError: [Errno 107] Transport endpoint is not connected
2020-05-20T20:15:37.5115476Z 
2020-05-20T20:15:37.5115893Z .venv/lib/python3.7/site-packages/irods/connection.py:263: OSError
```
Maybe you'd need to tolerate this OSErrror in the shutdown method ?

**Comment by kxk302** [comment]
@mvdbeek I am catching the OSError exception in shutdown() and also fixed another issue (poolside was missing in config for integration tests). Could u pls review when u have time? Thx

**Pull request merged**